tests/Fixer/Casing/VariableCaseFixerTest.php

```diff
@@ -0,0 +1,125 @@
+    {
+        return [
+            [
+                '<?php $testVariable = 2;',
+                '<?php $test_variable = 2;',
+            ],
+            [
+                '<?php $testVariable = 2; echo "hi $testVariable!";',
+                '<?php $test_variable = 2; echo "hi $test_variable!";',
+            ],
+            [
+                '<?php $testVariable = 2; echo "hi ${testVariable}!";',
+                '<?php $test_variable = 2; echo "hi ${test_variable}!";',
+            ],
+            [
+                '<?php $testVariable = 2; echo "hi {$testVariable}!";',
+                '<?php $test_variable = 2; echo "hi {$test_variable}!";',
+            ],
+            [
+                '<?php function foo_bar() { $testVariable = 2;}',
+                '<?php function foo_bar() { $test__variable = 2;}',
+            ],
+            [
+                '<?php echo $testModel->this_field;',
+                '<?php echo $test_model->this_field;',
+            ],
+            [
+                '<?php function f($barBaz, $file) { require $file;}',
+                '<?php function f($bar_baz, $file) { require $file;}',
+            ],
+        ];
+    }
+
+    /**
+     * @param string      $expected
+     * @param null|string $input
+     *
+     * @dataProvider provideSnakeCaseFixCases
+     */
+    public function testSnakeCaseFix($expected, $input = null)
+    {
+        $this->fixer->configure(['case' => VariableCaseFixer::SNAKE_CASE]);
+        $this->doTest($expected, $input);
+    }
+
+    public function provideSnakeCaseFixCases()
+    {
+        return [
+            [
+                '<?php $test_variable = 2;',
+                '<?php $testVariable = 2;',
+            ],
+            [
+                '<?php $abc_12_variable = 2;',
+                '<?php $abc12Variable = 2;',
+            ],
+            [
+                '<?php $abc_123_a_variable = 2;',
+                '<?php $abc123aVariable = 2;',
+            ],
+            [
+                '<?php function fooBar() { $test_variable = 2;}',
+                '<?php function fooBar() { $testVariable = 2;}',
+            ],
+            [
+                '<?php $test_variable = 2; echo "hi $test_variable!";',
+                '<?php $testVariable = 2; echo "hi $testVariable!";',
+            ],
+            [
+                '<?php $test_variable = 2; echo "hi ${test_variable}!";',
+                '<?php $testVariable = 2; echo "hi ${testVariable}!";',
+            ],
+            [
+                '<?php $test_variable = 2; echo "hi {$test_variable}!";',
+                '<?php $testVariable = 2; echo "hi {$testVariable}!";',
+            ],
+            [
+                '<?php echo $test_model->this_field;',
+                '<?php echo $testModel->this_field;',
+            ],
+            [
+                '<?php function f($bar_baz, $file) { require $file;}',
+                '<?php function f($barBaz, $file) { require $file;}',
```

Comment:
Why has this test been included? It just shows a broken example?

Reply:
Can I get some more clarification? How is this broken? It is converting `$barBaz` to `$bar_baz` and not altering anything else, so I must be missing something.

Reply:
Suppose the required file has the following content:

```php
<?php

echo $bar_baz;
```

Reply:
@GrahamCampbell yes? Do you mean if the variable is created in a different file? If so, I think that's just part of it being risky... but if it's run across the whole code base, it should fix it in both places.